### DISCUSSION TITLE
Wandb "step" axis definition

### DISCUSSION THREAD

**Discussion by adawne** [thu-ml/tianshou#1176] [Q&A]
Hi, I tried to log my experiment using the wandb logger with parameters as follows:
  ```
  logger = WandbLogger(
        train_interval=1,
        update_interval=1,
        info_interval=1,
        test_interval=1,
        save_interval=200,
        project=args.task,
        config=args,
    )
```
When I monitored my experiment, I realized something weird where the number of epochs executed is 8 epochs but the "step" in the graph already shows "800". How is the step calculated or defined? (My step_per_epoch parameter was set to 1000)
![epoch](https://github.com/user-attachments/assets/a97bfe47-add2-4b06-b64f-1f88d9bffef6)
![graph](https://github.com/user-attachments/assets/ff5c5c42-9c61-43c5-aa1c-9a1522478e51)

**Comment by MischaPanch** [comment]
@maxhuettenrauch could you have a look pls?

**Reply by maxhuettenrauch** [reply]
Without the whole script it's pretty hard to tell. It's worth noting though that "Step" is the wandb internal step counter and is incremented everytime a wandb.log call is made (compare here https://community.wandb.ai/t/what-is-the-w-b-sdk-step-counter-logging/4975). In order to relate the metric to your training progress, it's better to set a custom x-axis, for example to 'train/env_step'.
